### PR TITLE
feat(FooterBlock): migrate to TS

### PR DESCRIPTION
**Pull request by MkMan**
https://aesoponline.atlassian.net/browse/CON-464

### CODE REVIEW

File: src/components/FooterBlock/FooterBlock.tsx [hunk 1]
```diff
@@ -31,10 +31,7 @@ const FooterBlock = ({
   const listItemClassSet = cx(styles.listItem, listItemClassName);
 
   return (
-    <Hidden
-      isMedium={!isVisibleOnTabletAndMobile}
-      isSmall={!isVisibleOnTabletAndMobile}
-    >
+    <Hidden when={isVisibleOnTabletAndMobile ? [] : ['sm', 'md']}>
```
Comment (MkMan) [inline src/components/FooterBlock/FooterBlock.tsx:34]: The API for `Hidden` has change and it no longer has `isMedium` or `isSmall`. Instead they're combined into the `when` prop.

Is this correct correct, @zack76 ?

File: src/components/FooterBlock/FooterBlock.spec.tsx [hunk 2]
```diff
@@ -6,11 +6,7 @@ import { FooterBlockFixture } from './FooterBlock.fixture';
 describe('<FooterBlock />', () => {
   it('should render base component correctly', () => {
     const { container } = render(
-      <FooterBlock
-        copy={FooterBlockFixture.blockNoLink.copy}
-        theme="dark"
-        title={FooterBlockFixture.blockNoLink.title}
```
[the remark this answers is not in the document]
Comment (MkMan) [reply src/components/FooterBlock/FooterBlock.spec.tsx:12]: It's no longer a prop for the [component](https://github.com/aesop/aesop-gel/pull/581/files#diff-25ee9124b04c48570e5cb23df150c47d9e35e7d147481410b83a2655221608ee). It must have been removed at some point and the tests weren't updated to match that. TS picked up on it though 😎

**Review by kevin-ho87 — APPROVED** [review]

File: src/components/FooterBlock/FooterBlock.spec.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zack76) [reply src/components/FooterBlock/FooterBlock.spec.tsx:12]: must be my fault

**Review by zack76 — APPROVED** [review]

**Review by hassanraza0054 — APPROVED** [review]
~~Looks good to me! Just left that one question comment. Great if you could drop some context on that.~~

Thanks for the context! Good to go 🚀

File: src/components/FooterBlock/FooterBlock.spec.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (hassanraza0054) [reply src/components/FooterBlock/FooterBlock.spec.tsx:12]: All g! TS actually helping clean house! Very nice 👍

**Pull request merged**